src/lib.rs

```diff
@@ -112,30 +92,39 @@ macro_rules! impl_fft_with_opts_and_plan_for {
             reals: &mut [$precision],
             imags: &mut [$precision],
             opts: &Options,
-            planner: &mut $planner,
+            planner: &$planner,
         ) {
             assert!(reals.len() == imags.len() && reals.len().is_power_of_two());
             let n: usize = reals.len().ilog2() as usize;
 
-            let twiddles_re = &mut planner.twiddles_re;
-            let twiddles_im = &mut planner.twiddles_im;
+            let mut twiddles_re = planner.twiddles_re.clone();
+            let mut twiddles_im = planner.twiddles_im.clone();
 
             // We shouldn't be able to execute FFT if the # of twiddles isn't equal to the distance
             // between pairs
             assert!(twiddles_re.len() == reals.len() / 2 && twiddles_im.len() == imags.len() / 2);
 
+            match planner.direction {
+                Direction::Reverse => {
+                    for z_im in imags.iter_mut() {
+                        *z_im = -*z_im;
+                    }
+                }
+                _ => (),
+            }
+
```

Comment:
This requires a full pass over the twiddles, which is slow because it's memory-bottlenecked. Can we bake this into the twiddles themselves in the planner, instead of doing it over and over in the hot path?

Reply:
@Shnatsel 

This is pre-processing and post-processing that we utilize only in the case of inverse FFT. It was part of this [bug fix](https://github.com/QuState/PhastFT/pull/29#issue-2278013993) that addressed that the `Direction::Reverse` twiddle factors were not being properly generated.

This was already benchmarked, [here](https://github.com/astral4/fft-benchmark/tree/main?tab=readme-ov-file#results). Surprisingly, the overhead seems to be negligible/non-existent. 

Ideally, we have to look into the twiddle generation and figure out why we couldn't just directly apply the inverse FFT by using the reverse twiddle factors.

Reply:
In that case SGTM